src/wp-includes/rest-api/endpoints/class-wp-rest-font-faces-controller.php

```diff
@@ -0,0 +1,941 @@
+		$schema   = $this->get_item_schema()['properties']['font_face_settings']['properties'];
+
+		// Sanitize settings based on callbacks in the schema.
+		foreach ( $settings as $key => $value ) {
+			$sanitize_callback = $schema[ $key ]['arg_options']['sanitize_callback'];
+			$settings[ $key ]  = call_user_func( $sanitize_callback, $value );
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Retrieves a collection of font faces within the parent font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$font_family = $this->get_parent_font_family_post( $request['font_family_id'] );
+		if ( is_wp_error( $font_family ) ) {
+			return $font_family;
+		}
+
+		return parent::get_items( $request );
+	}
+
+	/**
+	 * Retrieves a single font face within the parent font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		// Check that the font face has a valid parent font family.
+		$font_family = $this->get_parent_font_family_post( $request['font_family_id'] );
+		if ( is_wp_error( $font_family ) ) {
+			return $font_family;
+		}
+
+		if ( (int) $font_family->ID !== (int) $post->post_parent ) {
+			return new WP_Error(
+				'rest_font_face_parent_id_mismatch',
+				/* translators: %d: A post id. */
+				sprintf( __( 'The font face does not belong to the specified font family with id of "%d".' ), $font_family->ID ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return parent::get_item( $request );
+	}
+
+	/**
+	 * Creates a font face for the parent font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_item( $request ) {
+		$font_family = $this->get_parent_font_family_post( $request['font_family_id'] );
+		if ( is_wp_error( $font_family ) ) {
+			return $font_family;
+		}
+
+		// Settings have already been decoded by ::sanitize_font_face_settings().
+		$settings    = $request->get_param( 'font_face_settings' );
+		$file_params = $request->get_file_params();
+
+		// Check that the necessary font face properties are unique.
+		$query = new WP_Query(
+			array(
+				'post_type'              => $this->post_type,
+				'posts_per_page'         => 1,
+				'title'                  => WP_Font_Utils::get_font_face_slug( $settings ),
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			)
+		);
+		if ( ! empty( $query->get_posts() ) ) {
```

Comment:
Same here, the constructor already calls `get_posts()`.

```suggestion
		if ( ! empty( $query->posts ) ) {
```

Reply:
I'm addressing this in this Gutenberg PR: https://github.com/WordPress/gutenberg/pull/58736. Once merged, I'll commit to this PR.